prov/gni/src/gnix_vc.c

```diff
@@ -121,7 +122,8 @@ static void __gnix_vc_pack_conn_req(char *sbuf,
 	memcpy(cptr, src_smsg_attr, sizeof(gni_smsg_attr_t));
 	cptr += sizeof(gni_smsg_attr_t);
 	memcpy(cptr, src_irq_cq_mhdl, sizeof(gni_mem_handle_t));
-
+	cptr += sizeof(gni_mem_handle_t);
+	memcpy(cptr, &caps, sizeof(uint64_t));
```

Comment:
The `len` check before these `memcpy` calls is wrong.  Actually It was wrong even before your change.